src/dbinterface/DBInterface.cpp

```diff
@@ -0,0 +1,127 @@
+
+#include "DBInterface.h"
+#include "../util/logger/Logger.h"
+
+Logger interface_logger;
+
+int DBInterface::finalize() {
+    return sqlite3_close(database);
+}
+
+typedef vector<vector<pair<string, string>>> table_type;
+
+static int callback(void *ptr, int argc, char **argv, char **azColName) {
+    int i;
+    table_type *dbResults = static_cast<table_type *>(ptr);
+    vector<pair<string, string>> results;
+
+    for (i = 0; i < argc; i++) {
+        results.push_back(make_pair(azColName[i], argv[i] ? argv[i] : "NULL"));
+        // printf("%s = %s\n", azColName[i], argv[i] ? argv[i] : "NULL");
+    }
+    dbResults->push_back(results);
+    return 0;
+}
+
+vector<vector<pair<string, string>>> DBInterface::runSelect(string query) {
+    char *zErrMsg = 0;
+    vector<vector<pair<string, string>>> dbResults;
+
+    if (sqlite3_exec(database, query.c_str(), callback, &dbResults, &zErrMsg) != SQLITE_OK) {
+        interface_logger.log("SQL Error: " + string(zErrMsg) + " " + query, "error");
+        sqlite3_free(zErrMsg);
+    } else {
+        interface_logger.log("Operation done successfully", "info");
```

Comment:
We need not log this. If the operation done successfully, we are anyway not interested of knowing that. We need to avoid generating too many unnecessary log lines.

Reply:
Ok